### PR TITLE
Configuring local registry (bsc#1205158)

### PR DESCRIPTION
**Pull request by tbazant**
fixes https://bugzilla.suse.com/show_bug.cgi?id=1205158

### CODE REVIEW

File: xml/deploy_bootstrap.xml [hunk 1]
```diff
@@ -527,215 +493,703 @@ Containers Module 15 SP3 x86_64 (Activated)
...
+            <step>
+              <para>
+                Modify <filename>/etc/environment</filename> and include the
+                <option>export GODEBUG=x509ignoreCN=0</option> option for
+                self-signed certificates. Ensure that existing configuration in
+                <filename>/etc/environment</filename> is not overwritten:
+              </para>
+<screen>salt '*' cmd.shell 'echo "export GODEBUG=x509ignoreCN=0" >>/etc/environment</screen>
+              <para>
+                In the current terminal run the following command:
```
Comment (dariavladykina) [inline xml/deploy_bootstrap.xml:801]:
```suggestion
                In the current terminal, run the following command:
```

File: xml/deploy_bootstrap.xml [hunk 2]
```diff
@@ -527,215 +493,703 @@ Containers Module 15 SP3 x86_64 (Activated)
...
-                  that you can delete images afterward with the <command>skopeo
-                  delete</command> command.
+                  In the event that &suse; has provided a PTF, you can also
+                  inspect the PTF using the path provided by &suse;.
                 </para>
+              </example>
+            </step>
+            <step>
+              <para>
+                Login to the secure private registry.
```
Comment (dariavladykina) [inline xml/deploy_bootstrap.xml:931]:
```suggestion
                Log in to the secure private registry.
```

File: xml/deploy_bootstrap.xml [hunk 3]
```diff
@@ -527,215 +493,703 @@ Containers Module 15 SP3 x86_64 (Activated)
...
- -e REGISTRY_COMPATIBILITY_SCHEMA1_ENABLED=true -d registry:2
+&prompt.root;podman login REG_HOST_FQDN:5000 -u REG_USERNAME -p REG_PASSWORD
+&prompt.root;podman login REG_HOST_FQDN:5000 -u root -p linux
 </screen>
-              </step>
-              <step>
+              <note>
                 <para>
-                  Test secure access to the registry:
+                  A insecure private registry does not require login.
```
Comment (dariavladykina) [inline xml/deploy_bootstrap.xml:939]:
```suggestion
                  An insecure private registry does not require login.
```

File: xml/deploy_bootstrap.xml [hunk 4]
```diff
@@ -527,215 +493,703 @@ Containers Module 15 SP3 x86_64 (Activated)
...
                 </para>
+              </note>
+            </step>
+            <step>
+              <para>
+                Pull images to local registry, push images to private registry.
```
Comment (dariavladykina) [inline xml/deploy_bootstrap.xml:945]:
```suggestion
                Pull images to the local registry, push images to the private
                registry.
```

File: xml/deploy_bootstrap.xml [hunk 5]
```diff
@@ -527,215 +493,703 @@ Containers Module 15 SP3 x86_64 (Activated)
...
-              <package>skopeo</package> package for that purpose. For more
-              details, refer to the manual page (<command>man 1
-              skopeo-sync</command>). Consider the following examples:
-            </para>
-            <example>
-              <title>Viewing manifest files</title>
+            </step>
+            <step>
+              <para>
+                Logout of the private registry:
```
Comment (dariavladykina) [inline xml/deploy_bootstrap.xml:991]:
```suggestion
                Log out of the private registry:
```

File: xml/deploy_bootstrap.xml [hunk 6]
```diff
@@ -527,215 +493,703 @@ Containers Module 15 SP3 x86_64 (Activated)
...
+            <step>
+              <para>
+                List repositories:
+              </para>
+<screen>&prompt.user;curl -sSk https://REG_HOST_FQDN:5000/v2/_catalog -u root:linux | jq</screen>
+            </step>
+            <step>
+              <para>
+                Use <command>skopeo</command> to inspect the private registry.
+                Note that you need to login to make the following commands
```
Comment (dariavladykina) [inline xml/deploy_bootstrap.xml:1010]:
```suggestion
                Note that you need to log in to make the following commands
```

File: xml/deploy_bootstrap.xml [hunk 7]
```diff
@@ -527,215 +493,703 @@ Containers Module 15 SP3 x86_64 (Activated)
...
+</screen>
+            </step>
+            <step>
+              <para>
+                Test pulling an image from another node.
+              </para>
+              <substeps>
+                <step>
+                  <para>
+                    Login to the private registry. Note that an insecure
```
Comment (dariavladykina) [inline xml/deploy_bootstrap.xml:1053]:
```suggestion
                    Log in to the private registry. Note that an insecure
```

File: xml/deploy_bootstrap.xml [hunk 8]
```diff
@@ -527,215 +493,703 @@ Containers Module 15 SP3 x86_64 (Activated)
...
+                  </para>
+<screen>
+&prompt.root;podman login REG_HOST_FQDN:5000 -u REG_CERTIFICATE.CRT \
+  -p REG_PRIVATE-KEY.KEY
+</screen>
+                </step>
+                <step>
+                  <para>
+                    Pull an image from the private registry. Optionally, can
+                    specify login credentials on the command line. For example:
```
Comment (dariavladykina) [inline xml/deploy_bootstrap.xml:1064]:
```suggestion
                    Pull an image from the private registry. Optionally, you can
                    specify login credentials on the command line. For example:
```

File: xml/deploy_bootstrap.xml [hunk 9]
```diff
@@ -527,215 +493,703 @@ Containers Module 15 SP3 x86_64 (Activated)
...
+&prompt.root;podman pull REG_HOST_FQDN:5000/ses/7.1/ceph/ceph:latest
+&prompt.root;podman pull --creds= REG_USERNAME:REG_PASSWORD \
+  REG_HOST_FQDN:5000/ses/7.1/ceph/ceph:latest
+&prompt.root;podman pull --cert-dir=/etc/containers/certs.d \
+  REG_HOST_FQDN:5000/ses/7.1/ceph/ceph:latest
+</screen>
+                </step>
+                <step>
+                  <para>
+                    Logout of private registry when done:
```
Comment (dariavladykina) [inline xml/deploy_bootstrap.xml:1076]:
```suggestion
                    Log out of the private registry when done:
```

File: xml/deploy_bootstrap.xml [hunk 10]
```diff
@@ -527,215 +493,703 @@ Containers Module 15 SP3 x86_64 (Activated)
...
+                  </para>
+<screen>&prompt.root;podman logout REG_HOST_FQDN:5000</screen>
+                </step>
+              </substeps>
+            </step>
+          </procedure>
+        </sect4>
+        <sect4>
+          <title>Configure &ceph; to pull images from private registry</title>
```
Comment (dariavladykina) [inline xml/deploy_bootstrap.xml:1085]:
```suggestion
          <title>Configuring &ceph; to pull images from the
          private registry</title>
```

File: xml/deploy_bootstrap.xml [hunk 11]
```diff
@@ -527,215 +493,703 @@ Containers Module 15 SP3 x86_64 (Activated)
...
+  REG_HOST_FQDN:5000/ses/7.1/ceph/prometheus-node-exporter:1.1.2
+&prompt.cephuser;ceph config set mgr mgr/cephadm/container_image_prometheus \
+  REG_HOST_FQDN:5000/ses/7.1/ceph/prometheus-server:2.32.1
+&prompt.cephuser;ceph config set mgr mgr/cephadm/container_image_snmp_gateway \
+  REG_HOST_FQDN:5000/ses/7.1/ceph/prometheus-snmp_notifier:1.2.1
+</screen>
+            </step>
+            <step>
+              <para>
+                Double check if there are other custom configurations that need
```
Comment (dariavladykina) [inline xml/deploy_bootstrap.xml:1142]:
```suggestion
                Double-check if there are other custom configurations that need
```

File: xml/deploy_bootstrap.xml [hunk 12]
```diff
@@ -527,215 +493,703 @@ Containers Module 15 SP3 x86_64 (Activated)
...
+                to be modified or removed and if the correct configuration was
+                applied:
+              </para>
+<screen>&prompt.cephuser;ceph config dump | grep container_image</screen>
+            </step>
+            <step>
+              <para>
+                All the daemons need to be redeployed to ensure correct images
+                are being used:
```
Comment (dariavladykina) [inline xml/deploy_bootstrap.xml:1151]:
```suggestion
                All the daemons need to be redeployed to ensure the correct
                images are being used:
```

File: xml/deploy_bootstrap.xml [hunk 13]
```diff
@@ -527,215 +493,703 @@ Containers Module 15 SP3 x86_64 (Activated)
...
+                  Monitor the redeployment by running the following command:
+                </para>
+<screen>&prompt.cephuser;ceph -W cephadm</screen>
+              </tip>
+            </step>
+            <step>
+              <para>
+                After the upgrade is complete, validate that the daemons have
+                been updated. Note the <literal>VERSION</literal> and
+                <literal>IMAGE ID</literal> values in right columns of the
```
Comment (dariavladykina) [inline xml/deploy_bootstrap.xml:1168]:
```suggestion
                <literal>IMAGE ID</literal> values in the right columns of the
```

**Review by dariavladykina — COMMENTED** [review]
Hi, please see some suggestions, thanks!

**Review by tbazant — COMMENTED** [review]
Thank you @dariavladykina , excellent language review!

**Comment by tbazant** [comment]
> Hi, please see some suggestions, thanks!

i've just addressed all of them, thanks

File: xml/deploy_bootstrap.xml [hunk 14]
```diff
@@ -527,215 +493,706 @@ Containers Module 15 SP3 x86_64 (Activated)
...
+                Start the insecure registry by creating the necessary
+                directory, for example, <filename>/var/lib/registry</filename>
+                and starting the registry with the <command>podman</command>
+                command:
+              </para>
 <screen>
 &prompt.root;mkdir -p /var/lib/registry
-&prompt.root;podman run --privileged -d --name registry \
- -p <option>REG_HOST_PORT</option>:5000 -v /var/lib/registry:/var/lib/registry \
+&prompt.root;podman run --privileged -d --name registry --net=host \
```
Comment (tbazant) [inline xml/deploy_bootstrap.xml:564]:
```suggestion
&prompt.root;podman run --privileged -d --name registry \
```

File: xml/deploy_bootstrap.xml [hunk 15]
```diff
@@ -527,215 +493,706 @@ Containers Module 15 SP3 x86_64 (Activated)
...
+              </para>
+<screen>
+&prompt.cephuser;ceph-salt config ls /containers
+o- containers .......................................................... [...]
+o- registries_conf ................................................. [enabled]
+| o- registries ...................................................... [empty]
+o- registry_auth ....................................................... [...]
+o- password .......................................................... [linux]
+o- registry ....................................... somehost.example.com:5000]
+o- username ........................................................... [root]
```
Comment (tbazant) [inline xml/deploy_bootstrap.xml:830]:
```suggestion
o- password .......................................................... [REG_PWD]
o- registry ....................................... somehost.example.com:5000]
o- username ........................................................... [REG_-USER]
```

File: xml/deploy_bootstrap.xml [hunk 16]
```diff
@@ -527,215 +493,706 @@ Containers Module 15 SP3 x86_64 (Activated)
...
- -e "REGISTRY_AUTH=htpasswd" \
- -e "REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm" \
- -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
- -v /var/lib/registry/certs:/certs:z \
- -e "REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt" \
- -e "REGISTRY_HTTP_TLS_KEY=/certs/domain.key" \
- -e REGISTRY_STORAGE_DELETE_ENABLED=true \
- -e REGISTRY_COMPATIBILITY_SCHEMA1_ENABLED=true -d registry:2
+&prompt.root;podman login REG_HOST_FQDN:5000 -u REG_USERNAME -p REG_PASSWORD
+&prompt.root;podman login REG_HOST_FQDN:5000 -u root -p linux
```
Comment (tbazant) [inline xml/deploy_bootstrap.xml:936]:
```suggestion
```

File: xml/deploy_bootstrap.xml [hunk 17]
```diff
@@ -527,215 +493,706 @@ Containers Module 15 SP3 x86_64 (Activated)
...
+              <para>
+                List images with <command>podman images</command> or
+                <command>podman images | sort</command>.
+              </para>
+            </step>
+            <step>
+              <para>
+                List repositories:
+              </para>
+<screen>&prompt.user;curl -sSk https://REG_HOST_FQDN:5000/v2/_catalog -u root:linux | jq</screen>
```
Comment (tbazant) [inline xml/deploy_bootstrap.xml:1007]:
```suggestion
<screen>&prompt.user;curl -sSk https://REG_HOST_FQDN:5000/v2/_catalog -u <replaceable>REG_USER</replaceable>:<replaceable>REG_PWD</replaceable> | jq</screen>
```

File: xml/deploy_bootstrap.xml [hunk 18]
```diff
@@ -527,215 +493,706 @@ Containers Module 15 SP3 x86_64 (Activated)
...
- -v /var/lib/registry/auth:/auth:z \
- -e "REGISTRY_AUTH=htpasswd" \
- -e "REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm" \
- -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
- -v /var/lib/registry/certs:/certs:z \
- -e "REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt" \
- -e "REGISTRY_HTTP_TLS_KEY=/certs/domain.key" \
- -e REGISTRY_STORAGE_DELETE_ENABLED=true \
- -e REGISTRY_COMPATIBILITY_SCHEMA1_ENABLED=true -d registry:2
+&prompt.root;podman login REG_HOST_FQDN:5000 -u REG_USERNAME -p REG_PASSWORD
```
Comment (tbazant) [inline xml/deploy_bootstrap.xml:935]:
```suggestion
&prompt.root;podman login REG_HOST_FQDN:5000 -u <replaceable>REG_USER</replaceable> -p <replaceable>REG_PWD</replaceable>
```

**Review by tbazant — COMMENTED** [review]
Changes suggested by Boyd

**Pull request merged**